samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                            _device->CreateRenderTargetView(renderTarget, pDesc: null, rtvHandle);
+                            rtvHandle.Offset(1, _rtvDescriptorSize);
+                        }
+                        _renderTargets[unchecked((int)n)] = renderTarget;
+                    }
+                }
+
+                fixed (ID3D12CommandAllocator** commandAllocator = &_commandAllocator)
+                {
+                    iid = IID_ID3D12CommandAllocator;
+                    ThrowIfFailed(nameof(ID3D12Device.CreateRenderTargetView), _device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, &iid, (void**)commandAllocator));
+                }
+            }
+            finally
+            {
+                if (debugController != null)
+                {
+                    debugController->Release();
+                }
+
+                if (factory != null)
+                {
+                    factory->Release();
+                }
+
+                if (adapter != null)
+                {
+                    adapter->Release();
+                }
+
+                if (swapChain != null)
+                {
+                    swapChain->Release();
+                }
+            }
+        }
+
+        // Load the sample assets.
+        private void LoadAssets()
+        {
+            Guid iid;
+            ID3DBlob* signature = null;
+            ID3DBlob* error = null;
+            ID3DBlob* vertexShader = null;
+            ID3DBlob* pixelShader = null;
+
+            try
+            {
+                // Create a root signature consisting of a descriptor table with a single CBV.
+                {
+                    D3D12_FEATURE_DATA_ROOT_SIGNATURE featureData = new D3D12_FEATURE_DATA_ROOT_SIGNATURE {
+
+                        // This is the highest version the sample supports. If CheckFeatureSupport succeeds, the HighestVersion returned will not be greater than this.
+                        HighestVersion = D3D_ROOT_SIGNATURE_VERSION_1_1
+                    };
+
+                    if (FAILED(_device->CheckFeatureSupport(D3D12_FEATURE.D3D12_FEATURE_ROOT_SIGNATURE, &featureData, (uint)sizeof(D3D12_FEATURE_DATA_ROOT_SIGNATURE))))
+                    {
+                        featureData.HighestVersion = D3D_ROOT_SIGNATURE_VERSION_1_0;
+                    }
+
+                    var ranges = new D3D12_DESCRIPTOR_RANGE1[1];
+                    const int RootParametersCount = 1;
+                    var rootParameters = new D3D12_ROOT_PARAMETER1[RootParametersCount];
+
+                    ranges[0].Init(D3D12_DESCRIPTOR_RANGE_TYPE.D3D12_DESCRIPTOR_RANGE_TYPE_CBV, 1, 0, 0, D3D12_DESCRIPTOR_RANGE_FLAGS.D3D12_DESCRIPTOR_RANGE_FLAG_DATA_STATIC);
+                    fixed (D3D12_DESCRIPTOR_RANGE1* pRanges = &ranges[0])
+                    {
+                        rootParameters[0].InitAsDescriptorTable(1, pRanges, D3D12_SHADER_VISIBILITY.D3D12_SHADER_VISIBILITY_VERTEX);
+                    }
```

Comment:
Changing the above to a stackalloc and adding a `using static TerraFX.Interop.D3D12_SHADER_VISIBILITY` will allow this to become
```suggestion
                        rootParameters[0].InitAsDescriptorTable(RangesCount, ranges, D3D12_SHADER_VISIBILITY_VERTEX);
```